vpc_endpoints.tf

```diff
@@ -0,0 +1,226 @@
+resource "aws_vpc_endpoint" "logs" {
+  provider = aws.provisionassessment
+
+  private_dns_enabled = true
+  security_group_ids = [
+    aws_security_group.cloudwatch.id,
+  ]
+  service_name      = "com.amazonaws.${var.aws_region}.logs"
+  tags              = var.tags
+  vpc_endpoint_type = "Interface"
+  vpc_id            = aws_vpc.assessment.id
+}
+resource "aws_vpc_endpoint" "monitoring" {
+  provider = aws.provisionassessment
+
+  private_dns_enabled = true
+  security_group_ids = [
+    aws_security_group.cloudwatch.id,
+  ]
+  service_name      = "com.amazonaws.${var.aws_region}.monitoring"
+  tags              = var.tags
+  vpc_endpoint_type = "Interface"
+  vpc_id            = aws_vpc.assessment.id
+}
+
+# Associate the STS interface endpoint with the private subnets
+resource "aws_vpc_endpoint_subnet_association" "sts" {
+  provider = aws.provisionassessment
+
+  # Normally we would use a for_each here, but the private subnets are
+  # _all in the same AZ_ in this case.  Why even have multiple subnets
+  # if you're not going to spread them around?  Harumph!
+  # for_each = toset(var.private_subnet_cidr_blocks)
+
+  # subnet_id       = aws_subnet.private[each.value].id
+  subnet_id       = aws_subnet.private[var.private_subnet_cidr_blocks[0]].id
+  vpc_endpoint_id = aws_vpc_endpoint.sts.id
+}
+
+# Associate the SSM interface endpoints with the private subnets
+#
+# Note that SSM requires several other endpoints to function properly.
+# See here for more details:
+# https://docs.aws.amazon.com/systems-manager/latest/userguide/setup-create-vpc.html#sysman-setting-up-vpc-create
+resource "aws_vpc_endpoint_subnet_association" "ssm" {
+  provider = aws.provisionassessment
+
+  # Normally we would use a for_each here, but the private subnets are
+  # _all in the same AZ_ in this case.  Why even have multiple subnets
+  # if you're not going to spread them around?  Harumph!
+  # for_each = toset(var.private_subnet_cidr_blocks)
+
+  # subnet_id       = aws_subnet.private[each.value].id
+  subnet_id       = aws_subnet.private[var.private_subnet_cidr_blocks[0]].id
+  vpc_endpoint_id = aws_vpc_endpoint.ssm.id
+}
+resource "aws_vpc_endpoint_subnet_association" "ec2" {
+  provider = aws.provisionassessment
+
+  # Normally we would use a for_each here, but the private subnets are
+  # _all in the same AZ_ in this case.  Why even have multiple subnets
+  # if you're not going to spread them around?  Harumph!
+  # for_each = toset(var.private_subnet_cidr_blocks)
+
+  # subnet_id       = aws_subnet.private[each.value].id
+  subnet_id       = aws_subnet.private[var.private_subnet_cidr_blocks[0]].id
+  vpc_endpoint_id = aws_vpc_endpoint.ec2.id
+}
+resource "aws_vpc_endpoint_subnet_association" "ec2messages" {
+  provider = aws.provisionassessment
+
+  # Normally we would use a for_each here, but the private subnets are
+  # _all in the same AZ_ in this case.  Why even have multiple subnets
+  # if you're not going to spread them around?  Harumph!
```

Comment:
[Harumph!](https://www.youtube.com/watch?v=LmnaeU0k80I)

Reply:
Also this bit from "Blazing Saddles":
[![Harumph](https://img.youtube.com/vi/Yt3GBlVjUd0/0.jpg)](https://www.youtube.com/watch?v=Yt3GBlVjUd0)